src/Foundatio.Repositories.Elasticsearch/Repositories/ElasticReadOnlyRepositoryBase.cs

```diff
@@ -389,6 +385,10 @@ public abstract class ElasticReadOnlyRepositoryBase<T> : ISearchableReadOnlyRepo
             var nextPageFunc = ((IGetNextPage<TResult>)result).GetNextPageFunc;
             ((IGetNextPage<TResult>)result).GetNextPageFunc = null;
             await SetCachedQueryResultAsync(options, result, cacheSuffix: cacheSuffix).AnyContext();
+            
+            if (IsCacheEnabled && options.ShouldUseCache() && options.GetConsistency(DefaultConsistency) == Consistency.Immediate)
+                await AddDocumentsToCacheAsync((ICollection<T>)result.Documents, options, false);
```

Comment:
Because this query result is already potentially getting cached as a whole result. It seems strange to me to send a potentially large list of documents through the AddDocumentsToCache that are all dirty reads to be added individually to cache. I guess it could be confusing that it doesn't happen, but I think I'd rather argue that documents that go through find and potentially return large results don't go through the AddDocumentsToCache at all even if it's not a dirty read.

Reply:
This is actually bad because we are already caching the entire result with the specified cache key. Calling this method could interfere with that and also could flood your cache with tons of docs being added by id. I'm going to remove this.